formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/aws/S3Client.scala

```diff
@@ -86,14 +97,19 @@ object S3 extends S3Client with LazyLogging {
   override def writeSuccessResult(
     initiationId: String,
     results: List[FormstackSubmissionQuestionAnswer],
-    config: PerformSarLambdaConfig): Either[Throwable, S3WriteSuccess] = {
-    val resultsPath = s"${config.resultsPath}/$initiationId/completed"
-    val filePath = if (results.nonEmpty) {
-      logger.info("Writing SAR result to s3.")
-      s"$resultsPath/formstackSarResponse"
-    } else {
-      logger.info(s"No results found for request $initiationId. Creating NoResultsFoundForUser object.")
-      s"$resultsPath/noResultsFoundForUser"
+    requestType: BatonRequestType,
+    config: PerformLambdaConfig): Either[Throwable, S3WriteSuccess] = {
+
+    val filePath = requestType match {
+      case SAR if results.nonEmpty =>
```

Comment:
we can have only one case here. We check for case SAR and use the results empty check to populate the last Option that goes into the generateResultsPath method.

Reply:
or better still move this logic to generateResultsPath method.